pkgs/applications/emulators/cdemu/base.nix

```diff
@@ -1,12 +1,16 @@
-{ pname, version, pkgSha256 }:
```

Comment:
I do not think this is how `callPackage` is intended to be used. IMO it is intended to auto-wire (in dependency injection parlance) packages from `pkgs` set (or other scope) and to allow `override`ing the auto-wired arguments.

Mixing it with parameters and attributes configuring the derivation is messy, and, in case of arguments with default value, is prone to future breakage. See also https://github.com/NixOS/nixpkgs/issues/309892

Additionally, calling `callPackage` inside `callPackage` will make it impossible to `override` the dependencies.

Reply:
Judging by the state of the issue you linked, the proliferation of non - auto-wired default-parameters throughout nixpkgs, and the lack of established alternatives, I'd still judge this to be the intended use of callPackage, despite being dangeroous and even if just intended by toleration.

Curious though, how in this light, the previous solution with two lambdas was actually "more correct" ...

Reply:
I avoided the nested callPackage in latest commit